pddl/Pddl.g4

```diff
@@ -1,127 +1,208 @@
 /**
- * PDDL grammar for ANTLR v3
+ * PDDL grammar for ANTLR v4
+ *
  * Zeyn Saigol
  * School of Computer Science
  * University of Birmingham
  *
- * $Id: Pddl.g 120 2008-10-02 14:59:50Z zas $
+ * modified for v4 and made it language agnostic by
+ * Hernán M. Foffani
+ * Now part of the project
+ * https://bitbucket.org/hfoffani/pddl-parser
+ *
+ *
+ * Modified to add Multi-Agent PDDL (MAPDDL),
+ * following most of the specification by Kovacs, 2012.
+ * https://home.mit.bme.hu/~dkovacs/pubs/d.l.kovacs_2012_ICAPS-WIPC.pdf
+ * Agents have the capability (in this implementation) to carry out
+ * whichever actions they please, future work sees the addition of
+ * agents to the goal condition.
+ * Stewart Anderson 2024-03-29
+ * (stewart.anderson@abdn.ac.uk)
+ *
+ * This file carries the same license from the original.
  */
 
-// $antlr-format alignTrailingComments true, columnLimit 150, minEmptyLines 1, maxEmptyLinesToKeep 1, reflowComments false, useTab false
-// $antlr-format allowShortRulesOnASingleLine false, allowShortBlocksOnASingleLine true, alignSemicolons hanging, alignColons hanging
+ // $antlr-format alignTrailingComments true, columnLimit 150, minEmptyLines 1, maxEmptyLinesToKeep 1, reflowComments false, useTab false
+ // $antlr-format allowShortRulesOnASingleLine false, allowShortBlocksOnASingleLine true, alignSemicolons hanging, alignColons hanging
 
 grammar Pddl;
 
+tokens {
+    DOMAIN,
+    DOMAIN_NAME,
+    REQUIREMENTS,
+    TYPES,
+    EITHER_TYPE,
+    CONSTANTS,
+    FUNCTIONS,
+    PREDICATES,
+    ACTION,
+    DURATIVE_ACTION,
+    PROBLEM,
+    PROBLEM_NAME,
+    PROBLEM_DOMAIN,
+    OBJECTS,
+    INIT,
+    FUNC_HEAD,
+    PRECONDITION,
+    EFFECT,
+    AND_GD,
+    OR_GD,
+	NOT_GD,
+	IMPLY_GD,
+	EXISTS_GD,
+	FORALL_GD,
+	COMPARISON_GD,
+	AND_EFFECT,
+	FORALL_EFFECT,
+	WHEN_EFFECT,
+	ASSIGN_EFFECT,
+	NOT_EFFECT,
+	PRED_HEAD,
+	GOAL,
+	BINARY_OP,
+	UNARY_MINUS,
+	INIT_EQ,
+	INIT_AT,
+	NOT_PRED_INIT,
+	PRED_INST,
+	PROBLEM_CONSTRAINT,
+	PROBLEM_METRIC
+}
+
+
 /************* Start of grammar *******************/
-pddlDoc
-    : (domain | problem) EOF
-    ;
+
+pddlDoc : domain | problem;
```

Comment:
All Antlr4 grammars need an EOF-terminated start rule. Add EOF back in so the tester can find the rule.

Reply:
Fixed.